service/validator.go

```diff
@@ -49,12 +49,14 @@ func NewValidator(audienceConfig config.AudienceConfig, opts ...ValidatorOption)
 		audience:           audienceConfig,
 	}
 
-	serviceValidator.issuer = serviceValidator.realmURL()
-
 	for _, opt := range opts {
 		opt(serviceValidator)
 	}
 
+	if len(serviceValidator.issuer) == 0 {
```

Comment:
fyi: checking if the string is empty is required so that mocking the issuer with `withIssuer` still works